--- mdadm-3.3rc1/udev-md-raid-assembly.rules.udev	2013-04-22 17:24:22.000000000 +0200
+++ mdadm-3.3rc1/udev-md-raid-assembly.rules	2013-06-28 14:59:58.092070982 +0200
@@ -4,16 +4,18 @@
 
 SUBSYSTEM!="block", GOTO="md_inc_end"
 
+# In Fedora we handle the raid components in 65-md-incremental.rules so that
+# we can do things like honor anaconda command line options and such
 # handle potential components of arrays (the ones supported by md)
-ENV{ID_FS_TYPE}=="ddf_raid_member|isw_raid_member|linux_raid_member", GOTO="md_inc"
-GOTO="md_inc_end"
+#ENV{ID_FS_TYPE}=="ddf_raid_member|isw_raid_member|linux_raid_member", GOTO="md_inc"
+#GOTO="md_inc_end"
 
-LABEL="md_inc"
+#LABEL="md_inc"
 
 # remember you can limit what gets auto/incrementally assembled by
 # mdadm.conf(5)'s 'AUTO' and selectively whitelist using 'ARRAY'
-ACTION=="add", RUN+="/sbin/mdadm --incremental $devnode --offroot"
-ACTION=="remove", ENV{ID_PATH}=="?*", RUN+="/sbin/mdadm -If $name --path $env{ID_PATH}"
-ACTION=="remove", ENV{ID_PATH}!="?*", RUN+="/sbin/mdadm -If $name"
+#ACTION=="add", RUN+="/sbin/mdadm --incremental $devnode --offroot"
+#ACTION=="remove", ENV{ID_PATH}=="?*", RUN+="/sbin/mdadm -If $name --path $env{ID_PATH}"
+#ACTION=="remove", ENV{ID_PATH}!="?*", RUN+="/sbin/mdadm -If $name"
 
 LABEL="md_inc_end"
